src/VisualStudio/Core/Impl/CodeModel/NodeKeyValidation.cs

```diff
@@ -17,14 +17,17 @@ public NodeKeyValidation()
 
         public void AddProject(AbstractProject project)
         {
-            var provider = (IProjectCodeModelProvider)project;
-            var fcms = provider.ProjectCodeModel.GetCachedFileCodeModelInstances();
-
-            foreach (var fcm in fcms)
+            var provider = project as IProjectCodeModelProvider;
+            if (provider != null)
```

Comment:
@jaredpar, ```IProjectCodeModelProvider``` is actually just an abstraction that allows us to push our CodeModel stuff into into another assembly. It is wholly ours.

@CyrusNajmabadi, when/why did TypeScript start using our internal project system bits? I didn't know that was a thing. I though TypeScript didn't have a formal project system.